test/gtest/ucs/test_datatype.cc

```diff
@@ -1356,3 +1359,215 @@ UCS_MT_TEST_F(test_datatype_ptr_map_safe, safe_put, num_threads)
         }
     }
 }
+
+
+class test_piecewise_func : public test_datatype {
+protected:
+    /* Represents piecewise function segment which consists of the linear
+     * function and the range on which this function is applicable.
+     * Both `start` and `end` is included to the range.
+     */
+    struct segment {
+        size_t            start;
+        size_t            end;
+        ucs_linear_func_t func;
+    };
+
+    static std::string get_segment_string(const segment &seg)
+    {
+        std::stringstream ss;
+        ss << "segment [" << seg.start << ", " << seg.end << "] : ";
+        ss << "func f(x) = " << seg.func.c << " + x*" << seg.func.m;
+
+        return ss.str();
+    }
+
+    static void print_piecewise_func(const ucs_piecewise_func_t &func)
+    {
+        size_t start = 0;
+        ucs_piecewise_segment_t *seg;
+
+        UCS_TEST_MESSAGE << "piecewise func:";
+        ucs_list_for_each(seg, &func.head, list) {
+            segment test_segment = {start, seg->end, seg->func};
+            UCS_TEST_MESSAGE << "\t" << get_segment_string(test_segment);
+            start = seg->end + 1;
+        }
+    }
```

Comment:
better add piecewise_func API to dump itself to a string buffer

Reply:
`ucs_linear_func_t` has the API to stringify it the UCP, because usually it requires to convert data to MB/s or microseconds. I think it worth adding such interface on the UCP level for piecewise function too since printing it in the raw representation can be less readable from the UCP perspective. But I would like to do that in a separate patch. 

BTW I think it worth adding C API like `ucs_piecewise_func_seg_foreach()` to allow UCP piecewise func string dumping API without accessing internal fields and it also should help to resolve [this comment](https://github.com/openucx/ucx/pull/9569/files#r1457435371), but using a different C API.

WDYT?

Reply:
ucs_piecewise_func_seg_foreach sounds good

Reply:
Done.